src/ViewModels/CkanGroupViewModel.js

```diff
@@ -159,40 +168,57 @@ CkanGroupViewModel.prototype._load = function() {
         return undefined;
     }
 
-    var url = cleanAndProxyUrl(this.application, this.url) + '/api/3/action/package_search?rows=100000&fq=' + encodeURIComponent(this.filterQuery);
-
     var that = this;
 
-    return loadJson(url).then(function(json) {
-        if (that.filterByWmsGetCapabilities) {
-            return when(filterResultsByGetCapabilities(that, json), function() {
-                populateGroupFromResults(that, json);
+    var allResults = {result: {results: []}};
+    var handleResults = function(json) {
+            allResults.result.results = allResults.result.results.concat(json.result.results);
```

Comment:
`when.all` resolves to an array, where each element in the array corresponds to the resolved value of the promise in the same position in the array passed to `when.all`.  So you can do this concatenation in the `.then` at the bottom of this `_load` function, and it would be a lot easier to understand what's going on, IMO.  Plus you'd get deterministic ordering.  Currently results are concatenated in whatever order the promises happen to resolve in.